cpp/src_prims/cuda_utils.cuh

```diff
@@ -119,26 +119,7 @@ HDI void swapVals(T &a, T &b) {
 }
 
 /** Device function to have atomic add support for older archs */
-#if __CUDA_ARCH__ < 600
```

Comment:
I understand that our min supported arch is sm_60, but I strongly prefer keeping this code. We already have spent effort adding this logic in ml-prims. This will help folks who are at arch < sm_60 to atleast compile from source.

Reply:
If it were free, I'd be fine with minimizing changes and keeping it in, but making this work for arch < sm60 will take additional work now. But given all the challenges with getting this large PR in, I don't see a big value in doing additional work to support an architecture that RAPIDS. Especially if we're considering libcu++ atomics in a future release, I would prefer not to invest time in a stopgap for an unsupported architecture that can also be replaced shortly.

I'd suggest we file and issue as a reminder to add libcu++ atomics in the next release but otherwise go with the simplified (sm60+ only) for now.

Reply:
```cpp
template <typename Type>
DI void myAtomicAdd(Type* address, Type val) {
  atomicAdd(address, val);
}

#if defined(__CUDA_ARCH__) && (__CUDA_ARCH__ < 600)
// Ref:
// http://on-demand.gputechconf.com/gtc/2013/presentations/S3101-Atomic-Memory-Operations.pdf
template <>
DI void myAtomicAdd(double *address, double val) {
  unsigned long long int *address_as_ull = (unsigned long long int *)address;
  unsigned long long int old = *address_as_ull, assumed;
  do {
    assumed = old;
    old = atomicCAS(address_as_ull, assumed,
                    __double_as_longlong(val + __longlong_as_double(assumed)));
  } while (assumed != old);
}
#endif
```

Just doing this from Divye's branch in this PR seems to compile for me without any issues! So, there's hardly any significant change we'll have to do in order to keep this logic around.

I took the liberty to push this change in this PR.

Reply:
BTW, I only updated `src_prims` to show this solution works. @divyegala please update all other occurences of `myAtomicAdd` with `raft::myAtomicAdd`.